From 2e5aa4eb46c812a093ce53afe22eab4554e2298c Mon Sep 17 00:00:00 2001
From: Bas Nieuwenhuizen <basni@chromium.org>
Date: Wed, 11 Dec 2019 12:37:42 +0100
Subject: [PATCH 1/3] radv: Disable extensions that are not supported by P CTS.

dEQP-VK.api.info.device.extensions complains about new KHR
extensions.

Change-Id: I5950e06faa1ea59e39f24ca8221ab0b2152210ae
---
 src/amd/vulkan/radv_extensions.py | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/src/amd/vulkan/radv_extensions.py b/src/amd/vulkan/radv_extensions.py
index a081e2da87a..e68931bcd3c 100644
--- a/src/amd/vulkan/radv_extensions.py
+++ b/src/amd/vulkan/radv_extensions.py
@@ -56,7 +56,7 @@ EXTENSIONS = [
     Extension('VK_KHR_bind_memory2',                      1, True),
     Extension('VK_KHR_create_renderpass2',                1, True),
     Extension('VK_KHR_dedicated_allocation',              1, True),
-    Extension('VK_KHR_depth_stencil_resolve',             1, True),
+    Extension('VK_KHR_depth_stencil_resolve',             1, False),
     Extension('VK_KHR_descriptor_update_template',        1, True),
     Extension('VK_KHR_device_group',                      1, True),
     Extension('VK_KHR_device_group_creation',             1, True),
@@ -76,35 +76,35 @@ EXTENSIONS = [
     Extension('VK_KHR_get_physical_device_properties2',   1, True),
     Extension('VK_KHR_get_surface_capabilities2',         1, 'RADV_HAS_SURFACE'),
     Extension('VK_KHR_image_format_list',                 1, True),
-    Extension('VK_KHR_imageless_framebuffer',             1, True),
+    Extension('VK_KHR_imageless_framebuffer',             1, False),
     Extension('VK_KHR_incremental_present',               1, 'RADV_HAS_SURFACE'),
     Extension('VK_KHR_maintenance1',                      1, True),
     Extension('VK_KHR_maintenance2',                      1, True),
     Extension('VK_KHR_maintenance3',                      1, True),
-    Extension('VK_KHR_pipeline_executable_properties',    1, True),
+    Extension('VK_KHR_pipeline_executable_properties',    1, False),
     Extension('VK_KHR_push_descriptor',                   1, True),
     Extension('VK_KHR_relaxed_block_layout',              1, True),
     Extension('VK_KHR_sampler_mirror_clamp_to_edge',      1, True),
     Extension('VK_KHR_sampler_ycbcr_conversion',          1, True),
     Extension('VK_KHR_shader_atomic_int64',               1, 'LLVM_VERSION_MAJOR >= 9'),
-    Extension('VK_KHR_shader_clock',                      1, True),
+    Extension('VK_KHR_shader_clock',                      1, False),
     Extension('VK_KHR_shader_draw_parameters',            1, True),
-    Extension('VK_KHR_shader_float_controls',             1, '!device->use_aco'),
-    Extension('VK_KHR_shader_float16_int8',               1, '!device->use_aco'),
-    Extension('VK_KHR_spirv_1_4',                         1, True),
+    Extension('VK_KHR_shader_float_controls',             1, False),
+    Extension('VK_KHR_shader_float16_int8',               1, False),
+    Extension('VK_KHR_spirv_1_4',                         1, False),
     Extension('VK_KHR_storage_buffer_storage_class',      1, True),
     Extension('VK_KHR_surface',                          25, 'RADV_HAS_SURFACE'),
     Extension('VK_KHR_surface_protected_capabilities',    1, 'RADV_HAS_SURFACE'),
     Extension('VK_KHR_swapchain',                        68, 'RADV_HAS_SURFACE'),
-    Extension('VK_KHR_timeline_semaphore',                2, 'device->rad_info.has_syncobj_wait_for_submit'),
-    Extension('VK_KHR_uniform_buffer_standard_layout',    1, True),
+    Extension('VK_KHR_timeline_semaphore',                2, False),
+    Extension('VK_KHR_uniform_buffer_standard_layout',    1, False),
     Extension('VK_KHR_variable_pointers',                 1, True),
     Extension('VK_KHR_wayland_surface',                   6, 'VK_USE_PLATFORM_WAYLAND_KHR'),
     Extension('VK_KHR_xcb_surface',                       6, 'VK_USE_PLATFORM_XCB_KHR'),
     Extension('VK_KHR_xlib_surface',                      6, 'VK_USE_PLATFORM_XLIB_KHR'),
     Extension('VK_KHR_multiview',                         1, True),
     Extension('VK_KHR_display',                          23, 'VK_USE_PLATFORM_DISPLAY_KHR'),
-    Extension('VK_KHR_8bit_storage',                      1, 'device->rad_info.chip_class >= GFX8 && !device->use_aco'),
+    Extension('VK_KHR_8bit_storage',                      1, False),
     Extension('VK_EXT_direct_mode_display',               1, 'VK_USE_PLATFORM_DISPLAY_KHR'),
     Extension('VK_EXT_acquire_xlib_display',              1, 'VK_USE_PLATFORM_XLIB_XRANDR_EXT'),
     Extension('VK_EXT_buffer_device_address',             1, True),
-- 
2.24.0.525.g8f36a354ae-goog

